lib/app/features/users/data/repositories/users_repository.dart

```diff
@@ -1,21 +1,24 @@
 import 'dart:convert';
 
 import 'package:dartz/dartz.dart';
-import 'package:penhas/app/core/error/failures.dart';
-import 'package:penhas/app/core/network/api_client.dart';
-import 'package:penhas/app/features/authentication/presentation/shared/map_exception_to_failure.dart';
-import 'package:penhas/app/features/users/data/models/user_detail_model.dart';
-import 'package:penhas/app/features/users/data/models/user_search_session_model.dart';
-import 'package:penhas/app/features/users/domain/entities/user_detail_entity.dart';
-import 'package:penhas/app/features/users/domain/entities/user_search_options.dart';
-import 'package:penhas/app/features/users/domain/entities/user_search_session_entity.dart';
-import 'package:penhas/app/shared/logger/log.dart';
+import '../../../../core/entities/valid_fiel.dart';
+import '../../../../core/error/failures.dart';
+import '../../../../core/network/api_client.dart';
+import '../../../authentication/presentation/shared/map_exception_to_failure.dart';
+import '../models/user_detail_model.dart';
+import '../models/user_search_session_model.dart';
+import '../../domain/entities/user_detail_entity.dart';
+import '../../domain/entities/user_search_options.dart';
+import '../../domain/entities/user_search_session_entity.dart';
+import '../../../../shared/logger/log.dart';
 
 abstract class IUsersRepository {
   Future<Either<Failure, UserDetailEntity>> profileDetail(String clientId);
   Future<Either<Failure, UserSearchSessionEntity>> search(
     UserSearchOptions option,
   );
+  Future<Either<Failure, void>> report(int clientId, String reason);
+  Future<Either<Failure, void>> block(int clientId);
```

Comment:
achei estranho que o Dart deixa qualquer coisa ser `void` na implementação rs
como a mensagem vem do back, pode deixar com `ValidField` aqui também

```suggestion
  Future<Either<Failure, ValidField>> report(int clientId, String reason);
  Future<Either<Failure, ValidField>> block(int clientId);
```